lib/smooth/core/io/i2c/I2CMasterDevice.cpp

```diff
@@ -204,7 +207,12 @@ namespace smooth::core::io::i2c
         // If we implemented slave clock-stretching then restore master timeout to its original value
         if (scl_timeout > 0)
         {
-            res |= i2c_set_timeout(port, orig_scl_timeout);
+            auto timeout_res = i2c_set_timeout(port, orig_scl_timeout);
+
+            if (timeout_res != ESP_OK)
+            {
+                log_error(timeout_res, address);
```

Comment:
I think I will remove the error checking here since if the read, write above failed we will exit and if it didn't fail then this write should succeed.  What do you think?

Reply:
No, keep it. There's no guarantee that an operation succeeds just because the one prior to it did. You never know when there is interference.